### PR TITLE
Rewrite sbin/sync-files.sh

### PR DESCRIPTION
**Pull request by freeformz**
### Relace s3cmd with the aws cli.

I've been having problems with s3cmd and I couldn't figure them out.
I've started to use the aws cli in other things, so trying to keep it
standard.

### Parallel shasums

It's been taking longer and longer for them, so kick them all off
and gather up the results via a fifo. This will max out your CPUs, but
that's a good thing.

### Switch to zsh

fifo/signal handling in bash is brain dead and wouldn't work reliably.
Works fine in zsh.

### Add Ability to ignore files

There are times where there is other work in progress that has added
files to the s3 bucket and I don't want to delete them from the bucket.
This option allows the user to ignore those files.

### CODE REVIEW

File: sbin/sync-files.sh [hunk 1]
```diff
@@ -28,52 +30,95 @@ cd "${td}"
...
     u="$(< "${jf}" jq -r '."'${f}'".URL')"
-    echo "Downloading: ${u}"
-    curl -J -o "${f}" -L --retry 15 --retry-delay 2 $u
+    curl -s -J -o "${f}" -L --retry 15 --retry-delay 2 $u 2>&1
   fi
 
-  s="$(< "${jf}" jq -r '."'${f}'".SHA')"
-  if [ ${#s} -eq 40 ]; then
-    sf="$(shasum "${f}" | cut -d \  -f 1)"
+  local sk="$(< "${jf}" jq -r '."'${f}'".SHA' 2>&1)"
```
Comment (jmorrell) [inline sbin/sync-files.sh]: I find bash kind of cryptic at the best of times, and that combined with the short filenames like `sk`, `jf`, `u` and `sf` here, make it difficult for me to follow the logic

**Review by jmorrell — APPROVED** [review]

File: sbin/sync-files.sh (hunk 1 above)
Comment (freeformz) [reply sbin/sync-files.sh]: sk == sha known
jf == json file
sf == sha file
u == url

Comment (freeformz) [reply sbin/sync-files.sh]: But yeah. I can rename them.

Comment (jmorrell) [reply sbin/sync-files.sh]: none of that was obvious to me 😅

**Pull request merged**